### ISSUE TITLE
[BUG]: libpython3.10.dylib (which was built for Mac OS X 11.7)

### ISSUE THREAD

**Issue by mohammali** [KernelWanderers/OCSysInfo#34]
**Describe the bug**
when I run the app on MacOS 10.15.7 (19H1615) Catalina, it is not working and showing below output:

```
Last login: Thu Sep 14 18:01:26 on ttys000
/Applications/OCSysInfo.app/Contents/MacOS/OCSysInfo ; exit;
➜  ~ /Applications/OCSysInfo.app/Contents/MacOS/OCSysInfo ; exit;
[13047] Error loading Python lib '/var/folders/xp/8wys6y8s38g1jgc5v_p9nfp00000gn/T/_MEIBxy3R8/libpython3.10.dylib': dlopen: dlopen(/var/folders/xp/8wys6y8s38g1jgc5v_p9nfp00000gn/T/_MEIBxy3R8/libpython3.10.dylib, 10): Symbol not found: _preadv
  Referenced from: /var/folders/xp/8wys6y8s38g1jgc5v_p9nfp00000gn/T/_MEIBxy3R8/libpython3.10.dylib (which was built for Mac OS X 11.7)
  Expected in: /usr/lib/libSystem.B.dylib
 in /var/folders/xp/8wys6y8s38g1jgc5v_p9nfp00000gn/T/_MEIBxy3R8/libpython3.10.dylib

[Process completed]
```

```
python3 main.py
Traceback (most recent call last):
  File "/Users/mxninja/dev/os/OCSysInfo/main.py", line 5, in <module>
    main()
  File "/Users/mxninja/dev/os/OCSysInfo/src/main.py", line 24, in main
    or os.environ.get("DEBUG", "0") == "1"
       ^^
UnboundLocalError: cannot access local variable 'os' where it is not associated with a value
```

**To Reproduce**
Just run the application or run the command `python3 main.py`
...

**Expected behavior**
Showing the hardware

**Screenshots**
From the APP
![image](https://github.com/KernelWanderers/OCSysInfo/assets/46114379/8b99fa8f-cd30-45b3-9abf-ad4ef354602f)

From the SC
![image](https://github.com/KernelWanderers/OCSysInfo/assets/46114379/77ffb194-cc1b-430e-a788-e5e2b739026d)

**Platform (please complete the following information):**
 - OS: MacOS Catalina
 - Version 10.15.7 (19H1615)

**Comment by Mahas1** [comment]
It seems we forgot to import `os` in `src/main.py`, can you try adding this line in line 3 in `src/main.py` and checking if it works??
```py
import os
```
You can clone the repo, add the line, install dependencies by `cd`ing into the project directory and running 
```sh
pip3 install -r requirements.txt
```
and running the program using
```sh
python3 main.py
```

**Comment by mohammali** [comment]
Now it is showing this

```
Traceback (most recent call last):
  File "/Users/mxninja/dev/os/OCSysInfo/main.py", line 5, in <module>
    main()
  File "/Users/mxninja/dev/os/OCSysInfo/src/main.py", line 30, in main
    from update.updater import OCSIUpdater
  File "/Users/mxninja/dev/os/OCSysInfo/update/updater.py", line 8, in <module>
    import regex
ModuleNotFoundError: No module named 'regex'
```

**Comment by mohammali** [comment]
I have installed the missing `regex` module and did what you say and it's working now
![image](https://github.com/KernelWanderers/OCSysInfo/assets/46114379/8b4900b9-eacd-4a0e-a3a9-71a27acf52df)

**Comment by Mahas1** [comment]
cc @kernel-dev

**Comment by kernel-dev** [comment]
Hi, @mohammali, thank you for reporting this issue. This was an oversight on my behalf – I’m not sure how, but it worked for me locally even with the borked imports. This should be fixed by #35.
